File: src/main/java/com/example/work/controller/request/body/AttendancesRequestBody.java
package com.example.work.controller.request.body;

import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@FieldDefaults(level= AccessLevel.PRIVATE)
public class AttendancesRequestBody {
    List<Attendance> attendances;
    Integer courseId;
    Date registeredTimestamp;
}

File: src/main/java/com/example/work/controller/request/body/AttendancesRequestBody.kt
package com.example.work.controller.request.body;

import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@FieldDefaults(level= AccessLevel.PRIVATE)
public class AttendancesRequestBody {

    @NotNull
    List<Attendance> attendances;
    @Min(1)
    Integer courseId;
    @NotNull
    Date registeredTimestamp;
}
